src/Core/HTTPApplication.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace SilverStripe\Core;
+
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+
+/**
+ * Invokes the HTTP application within an ErrorControlChain
+ */
+class HTTPApplication implements Application
```

Comment:
To be clear: I think that Kernel should definitely be part of core. I'm not sue about the Application interface; that might stay in core?

HTTPApplication isn't likely to be useful if you're using another controller system; in those cases you'd likely plug in the SilverStripe Kernel as a service another framework's app object.